### PR TITLE
vga peripheral added to zybo z7

### PR DESCRIPTION
**Pull request by 24x7fpga**
Added VGA support for zybo z7 and verified with lab 2_1_rectangle_ellipse_parabola.

### CODE REVIEW

File: boards/zybo_z7/board_specific.xdc [hunk 1]
```diff
@@ -121,25 +121,27 @@ set_property -dict { PACKAGE_PIN D18   IOSTANDARD LVCMOS33 } [get_ports { led[3]
...
-#set_property -dict { PACKAGE_PIN R14   IOSTANDARD LVCMOS33     } [get_ports { jd[3] }]; #IO_L6N_T0_VREF_34 Sch=jd_n[2]
-#set_property -dict { PACKAGE_PIN U14   IOSTANDARD LVCMOS33     } [get_ports { jd[4] }]; #IO_L11P_T1_SRCC_34 Sch=jd_p[3]
-#set_property -dict { PACKAGE_PIN U15   IOSTANDARD LVCMOS33     } [get_ports { jd[5] }]; #IO_L11N_T1_SRCC_34 Sch=jd_n[3]
-#set_property -dict { PACKAGE_PIN V17   IOSTANDARD LVCMOS33     } [get_ports { jd[6] }]; #IO_L21P_T3_DQS_34 Sch=jd_p[4]
-#set_property -dict { PACKAGE_PIN V18   IOSTANDARD LVCMOS33     } [get_ports { jd[7] }]; #IO_L21N_T3_DQS_34 Sch=jd_n[4]
+##Pmod Header JC                                                                                                                  
+set_property -dict { PACKAGE_PIN V15   IOSTANDARD LVCMOS33     } [get_ports { VGA_R[0] }]; #IO_L10P_T1_34 Sch=jc_p[1]   			 
+set_property -dict { PACKAGE_PIN W15   IOSTANDARD LVCMOS33     } [get_ports { VGA_R[1] }]; #IO_L10N_T1_34 Sch=jc_n[1]		     
+set_property -dict { PACKAGE_PIN T11   IOSTANDARD LVCMOS33     } [get_ports { VGA_R[2] }]; #IO_L1P_T0_34 Sch=jc_p[2]
```
Comment (yuri-panchul) [inline boards/zybo_z7/board_specific.xdc:127]: The general strategy we use for other Xilinx boards is to keep vendor's XDC files as is and change the connections in board_specific_top.sv. Can you change it back in XDC and just modify board_specific_top.sv in the following way: .hsync ( jd [5] )?

File: boards/zybo_z7/board_specific_top.sv [hunk 2]
```diff
@@ -56,6 +63,8 @@ module board_specific_top
     wire [ w_red     - 1:0] red;
     wire [ w_green   - 1:0] green;
     wire [ w_blue    - 1:0] blue;
+    
+    wire                    display_on;
```
Comment (yuri-panchul) [inline boards/zybo_z7/board_specific_top.sv:67]: Do we need display_on here? I don't think it is used for this board

File: boards/zybo_z7/board_specific_top.sv [hunk 3]
```diff
@@ -159,6 +168,7 @@ module board_specific_top
         .x             (   x               ),
         .y             (   y               ),
 
+        .display_on    (   display_on      ),
```
Comment (yuri-panchul) [inline boards/zybo_z7/board_specific_top.sv:171]: No, we should not introduce a new port unless it is used for something

File: labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv [hunk 4]
```diff
@@ -39,7 +39,8 @@ module lab_top
 
     input        [w_x     - 1:0] x,
     input        [w_y     - 1:0] y,
-
+    input                        display_on,
```
Comment (yuri-panchul) [inline labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv:42]: If we add  a new port here, we have to add a new port in all lab_top.sv files in all examples as well. It is better to revert the changes in lab_top, unless display_on is used for something

File: labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv [hunk 5]
```diff
@@ -100,14 +100,14 @@ module lab_top
 
         if (x_2 [w_x +: w_y] < y)  // Parabola
             blue = y [w_y - 1 -: w_blue];
-    end
+     end
```
Comment (yuri-panchul) [inline labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv:103]: Looks like the changes in this file is just spacing. It is better to check for the changes using "git diff" before submitting to avoid submitting spacing and tab changes

**Review by yuri-panchul — CHANGES_REQUESTED** [review]
Please see the comments

File: boards/zybo_z7/board_specific.xdc (hunk 1 above)
Comment (24x7fpga) [reply boards/zybo_z7/board_specific.xdc:127]: Will do. I will add two output ports gpio_JD and gpio_JC and make the connection. For the tm1638 the gpio connection in the xdc file is "gpio_JE[0]". I think this is what you mean. I will be make the changes accordingly.

File: boards/zybo_z7/board_specific_top.sv (hunk 2 above)
Comment (24x7fpga) [reply boards/zybo_z7/board_specific_top.sv:67]: We don't need it, I just added it in-case we may need to access it in future for some purpose. I can remove it if not required.

File: boards/zybo_z7/board_specific_top.sv (hunk 3 above)
Comment (24x7fpga) [reply boards/zybo_z7/board_specific_top.sv:171]: Will remove it.

File: boards/zybo_z7/board_specific_top.sv [hunk 6]
```diff
@@ -207,4 +217,39 @@ module board_specific_top
...
+            .CLK_MHZ     ( clk_mhz     ),
+            .PIXEL_MHZ   ( pixel_mhz   )
+        )
+        i_vga
+        (
+            .clk         ( clk         ),
+            .rst         ( rst         ),
+            .hsync       ( VGA_HS      ),
+            .vsync       ( VGA_VS      ),
+            .display_on  ( display_on  ),
```
[the remark this answers is not in the document]
Comment (24x7fpga) [reply boards/zybo_z7/board_specific_top.sv:243]: Will do.

File: labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv (hunk 4 above)
Comment (24x7fpga) [reply labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv:42]: Will remove it.

File: labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv (hunk 5 above)
Comment (24x7fpga) [reply labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv:103]: Will do.

File: boards/zybo_z7/board_specific.xdc (hunk 1 above)
Comment (24x7fpga) [reply boards/zybo_z7/board_specific.xdc:127]: Let me know if you want jd[7] or gpio_JD[7].

Comment (yuri-panchul) [reply boards/zybo_z7/board_specific.xdc:127]: It is generally better (when possible) to keep the signal name just like in the original XDC file from the vendor - https://github.com/Digilent/digilent-xdc

File: labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv [hunk 7]
```diff
@@ -39,7 +39,7 @@ module lab_top
 
     input        [w_x     - 1:0] x,
     input        [w_y     - 1:0] y,
-
+
```
Comment (yuri-panchul) [inline labs/2_graphics/2_1_rectangle_ellipse_parabola/lab_top.sv:42]: Can you remove this file from pull request?

**Review by yuri-panchul — CHANGES_REQUESTED** [review]
Is it possible to remove lab_top.sv file from pull request? I don't want to make and correct 1-2 space changes

**Comment by 24x7fpga** [comment]
Sure Yuri, will keep all this in mind for the next commit.
